krakend/content.md

```diff
@@ -0,0 +1,144 @@
+
+Your `Dockerfile` could look like this:
+
+```Dockerfile
+FROM %%IMAGE%%:<version>
+# NOTE: Avoid using :latest image on production. Stick to a major version instead.
+
+COPY krakend.json /etc/krakend/krakend.json
+
+# Check and test that the file is valid
+RUN krakend check -d -t -c /etc/krakend/krakend.json
+```
+
+If you want to manage your KrakenD configuration using multiple files and folders, reusing templates, and distributing the configuration amongst your teams, you can use the [flexible configuration (FC)](https://www.krakend.io/docs/configuration/flexible-config/). The following `Dockerfile` combines FC, the `krakend check` command, and a 2-step build.
+
+```Dockerfile
+FROM %%IMAGE%%:<version> as builder
+
+COPY krakend.tmpl .
+COPY config .
+
+# Save temporary output file to /tmp to avoid permission errors
+RUN FC_ENABLE=1 \
+    FC_OUT=/tmp/krakend.json \
+    FC_PARTIALS="/etc/krakend/partials" \
+    FC_SETTINGS="/etc/krakend/settings" \
+    FC_TEMPLATES="/etc/krakend/templates" \
+    krakend check -d -t -c krakend.tmpl
+
+# Copy the output file only and discard any other files
+FROM %%IMAGE%%:<version>
+COPY --from=builder /tmp/krakend.json .
+```
+Then build with `docker build -t my_krakend .`
+
+The configuration above assumes you have a folder structure like the following:
+```
+.
+├── config
+│   ├── partials
+│   ├── settings
+│   │   └── env.json
+│   └── templates
+│       └── some.tmpl
+├── Dockerfile
+└── krakend.tmpl
+```
+
+### Docker Compose example
+Finally, a simple `docker-compose` file to start KrakenD with your API would be:
+
+```yaml
+version: "3"
+services:
+  krakend:
+    image: %%IMAGE%%:<version>
+    ports:
+      - "8080:8080"
+    volumes:
+      - ./:/etc/krakend
+```
+
+And another one that uses the flexible configuration and a custom template filename (`my_krakend.tmpl`) on each start:
+
+```yaml
+version: "3"
+services:
+  krakend:
+    image: %%IMAGE%%:<version>
+    ports:
+      - "8080:8080"
+    volumes:
+      - ./:/etc/krakend
+    environment:
+      - FC_ENABLE=1
+      - FC_OUT=/tmp/krakend.json
+      - FC_PARTIALS="/etc/krakend/config/partials"
+      - FC_SETTINGS="/etc/krakend/config/settings/prod"
+      - FC_TEMPLATES="/etc/krakend/config/templates"
+    command:
+      command: ["krakend", "run", "-c", "krakend.tmpl", "-d"]
+```
+### Container permissions and commands
+All `krakend` commands are executed as `krakend` user (uid=1000) through `su-exec`, and the rest of the commands (e.g., `sh`) are executed as root.
+
+You can directly use sub-commands of `krakend` like `run`, `help`, `version`, `check`, `check-plugin` or `validate` as the entrypoint will add the `krakend` command automatically. For example, the following lines are equivalent:
+
+    docker run -it -p 8080:8080 -v $PWD:/etc/krakend/ %%IMAGE%% help
+    docker run -it -p 8080:8080 -v $PWD:/etc/krakend/ %%IMAGE%% krakend help
```

Comment:
The port is not necessary here since it runs the CLI right?

Reply:
Absolutely right. Removed. Also the volume.